bin/anthology/papers.py

```diff
@@ -74,10 +74,9 @@ def from_xml(xml_element, *args):
                 del paper.attrib["editor"]
             paper.attrib[key] = value
 
-        # Frontmatter title is the volume 'booktitle'
-        if paper.is_volume:
-            paper.attrib["xml_title"] = paper.attrib["xml_booktitle"]
-            paper.attrib["xml_title"].tag = "title"
+        # Create URL field if not present. But see https://github.com/acl-org/acl-anthology/issues/997.
+        if "url" not in paper.attrib:
+            paper.attrib["url"] = data.ANTHOLOGY_URL.format(paper.full_id)
 
```

Comment:
"url" is not defined in the XML when there is no PDF (an unfortunate ambiguity). However, in the code generation, we use "url" to build the page's self-link, canonical URL. So here, we add the field it's not present.

Reply:
1. Probably should use `infer_url(paper.full_id)` for consistency with how it's done when a `<url>` tag exists.
2. Why are the lines above this removed?

Reply:
On (2), I'm  not sure how that happened!

For (1), I'll use `infer_url`, but it feels a bit funny to me, since we aren't inferring it so much as building it (the path will always be relative).